### PR TITLE
refactor: randomise resource type

### PR DESCRIPTION
**Pull request by rebeccabrennan**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/AM-245

### CODE REVIEW

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java [hunk 1]
```diff
@@ -77,7 +79,7 @@ void setUp() {
     @Test
     void returnListOfPermissionsForRoleName() {
         Map<JsonPointer, Set<Permission>> accessRecord =
-            service.getRolePermissions(buildResource(serviceName, RESOURCE_TYPE, RESOURCE_NAME), ROLE_NAMES);
+            service.getRolePermissions(buildResource(serviceName, resourceType, RESOURCE_NAME), ROLE_NAMES);
```
Comment (swalker125) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: why is buildResource needed here. Can you not pass in resourceDefinition from the setUp?

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GrantAccessIntegrationTest.java [hunk 2]
```diff
@@ -44,8 +43,9 @@ void setUp() {
         resourceId = UUID.randomUUID().toString();
         accessorId = UUID.randomUUID().toString();
         MDC.put("caller", "Administrator");
+        String resourceType = UUID.randomUUID().toString();
         importerService.addResourceDefinition(
-            resourceDefinition = createResourceDefinition(serviceName, RESOURCE_TYPE, RESOURCE_NAME));
+            resourceDefinition = createResourceDefinition(serviceName, resourceType, RESOURCE_NAME));
```
Comment (swalker125) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GrantAccessIntegrationTest.java]: similar query to above

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/RevokeAccessIntegrationTest.java [hunk 3]
```diff
@@ -41,8 +40,9 @@ void setUp() {
         resourceId = UUID.randomUUID().toString();
         accessorId = UUID.randomUUID().toString();
         MDC.put("caller", "Administrator");
+        String resourceType = UUID.randomUUID().toString();
         importerService.addResourceDefinition(
-            resourceDefinition = createResourceDefinition(serviceName, RESOURCE_TYPE, RESOURCE_NAME));
+            resourceDefinition = createResourceDefinition(serviceName, resourceType, RESOURCE_NAME));
```
Comment (swalker125) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/RevokeAccessIntegrationTest.java]: similar query to above

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/base/PreconfiguredIntegrationBaseTest.java [hunk 4]
```diff
@@ -31,6 +30,7 @@ void populateDatabaseWithBasicDefinitions() {
         importerService.addService(serviceName);
         importerService.addRole(ROLE_NAME, RESOURCE, PUBLIC, ROLE_BASED);
         importerService.addRole(OTHER_ROLE_NAME, IDAM, PUBLIC, EXPLICIT);
-        importerService.addResourceDefinition(createResourceDefinition(serviceName, RESOURCE_TYPE, RESOURCE_NAME));
+        String resourceType = UUID.randomUUID().toString();
+        importerService.addResourceDefinition(createResourceDefinition(serviceName, resourceType, RESOURCE_NAME));
```
Comment (swalker125) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/base/PreconfiguredIntegrationBaseTest.java]: see above

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java (hunk 1 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: 👍

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java [hunk 5]
```diff
@@ -54,10 +53,11 @@
 
     @BeforeEach
     void setUp() {
+        String resourceType = UUID.randomUUID().toString();
```
[the remark this answers is not in the document]
Comment (rebeccabrennan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java]: if we do this though it does not make it clear that we are looking for a resource type?

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/base/PreconfiguredIntegrationBaseTest.java (hunk 4 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/base/PreconfiguredIntegrationBaseTest.java]: 👍

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/RevokeAccessIntegrationTest.java (hunk 3 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/RevokeAccessIntegrationTest.java]: 👍

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GrantAccessIntegrationTest.java (hunk 2 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GrantAccessIntegrationTest.java]: 👍

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java (hunk 1 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: 👍

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (rebeccabrennan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java]: 👍

**Pull request merged**